src/internal/modules/contact-entry/routes.js

```diff
@@ -0,0 +1,183 @@
+  },
+  getEnterNewDetails: {
+    // Route for displaying the list of existing contact addresses.
+    method: 'GET',
+    path: '/contact-entry/new/details',
+    handler: controllers.getDetailsController,
+    options: {
+      validate: {
+        query: {
+          sessionKey: Joi.string().uuid().required(),
+          form: Joi.string().optional()
+        }
+      }
+    }
+  },
+  postEnterPersonDetails: {
+    // Route for posting the name of a new person contact
+    method: 'POST',
+    path: '/contact-entry/new/details/person',
+    handler: controllers.postPersonDetailsController
+  },
+  getAfterAddressEntry: {
+    method: 'GET',
+    path: '/contact-entry/new/details/after-address-entry',
+    handler: controllers.getAfterAddressEntryController,
+    options: {
+      validate: {
+        query: {
+          sessionKey: Joi.string().uuid().required(),
+          form: Joi.string().optional()
+        }
+      }
+    }
+  },
+  postCompanySearch: {
+    // Route for posting the name or number of a new company contact
+    // It searches companies house for a matching company
+    method: 'POST',
+    path: '/contact-entry/new/details/company-search',
+    handler: controllers.postCompanySearchController
+  },
+  getSelectCompany: {
+    method: 'GET',
+    path: '/contact-entry/new/details/company-search/select-company',
+    handler: controllers.getSelectCompanyController,
+    options: {
+      pre: [
+        { method: preHandlers.searchForCompaniesInCompaniesHouse, assign: 'companiesHouseResults' }
+      ],
+      validate: {
+        query: {
+          sessionKey: Joi.string().uuid().required(),
+          form: Joi.string().optional()
+        }
+      }
+    }
+  },
+  postSelectCompany: {
+    method: 'POST',
+    path: '/contact-entry/new/details/company-search/select-company',
+    handler: controllers.postSelectCompanyController,
+    options: {
+      pre: [
+        { method: preHandlers.searchForCompaniesInCompaniesHouse, assign: 'companiesHouseResults' }
+      ]
+    }
+  },
+  getSelectCompanyAddress: {
+    method: 'GET',
+    path: '/contact-entry/new/details/company-search/select-company-address',
+    handler: controllers.getSelectCompanyAddressController,
+    options: {
+      pre: [
+        { method: preHandlers.returnCompanyAddressesFromCompaniesHouse, assign: 'companiesHouseAddresses' }
+      ],
+      validate: {
+        query: {
+          sessionKey: Joi.string().uuid().required(),
+          form: Joi.string().optional()
+        }
+      }
+    }
+  },
+  postSelectCompanyAddress: {
+    method: 'POST',
+    path: '/contact-entry/new/details/company-search/select-company-address',
+    handler: controllers.postSelectCompanyAddressController,
+    options: {
+      pre: [
+        { method: preHandlers.returnCompanyAddressesFromCompaniesHouse, assign: 'companiesHouseAddresses' }
+      ]
+    }
+  },
+  getSelectAddress: {
+    // Route for displaying the list of existing contact addresses.
```

Comment:
not sure this route is required. Would you not use the address entry plugin that has the form and route for selecting the address

Reply:
Discussed with Stephan + Agreed to remove this form.

Done in https://github.com/DEFRA/water-abstraction-ui/pull/1394/commits/d09f56c66369ce08e17f8fa8ea3815d38f84d653